x/dispensation/keeper/executors.go

```diff
@@ -2,25 +2,26 @@ package keeper
 
 import (
 	"fmt"
+
 	"github.com/Sifchain/sifnode/x/dispensation/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/cosmos/cosmos-sdk/x/bank"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 	"github.com/pkg/errors"
 )
 
 //CreateAndDistributeDrops creates new drop Records . These records are then used to facilitate distribution
 // Each Recipient and DropName generate a unique Record
-func (k Keeper) CreateDrops(ctx sdk.Context, output []bank.Output, name string) error {
+func (k Keeper) CreateDrops(ctx sdk.Context, output []banktypes.Output, name string) error {
 	for _, receiver := range output {
-		distributionRecord := types.NewDistributionRecord(name, receiver.Address, receiver.Coins, ctx.BlockHeight(), -1)
-		if k.ExistsDistributionRecord(ctx, name, receiver.Address.String()) {
-			oldRecord, err := k.GetDistributionRecord(ctx, name, receiver.Address.String())
+		distributionRecord := types.NewDistributionRecord(name, receiver.Address, receiver.Coins, sdk.NewInt(ctx.BlockHeight()), sdk.NewInt(1))
+		if k.ExistsDistributionRecord(ctx, name, receiver.Address) {
```

Comment:
the older value was -1 ,for a new record which means the record has not been distributed yet . -1 is an impossible height .
If we use 1 , which is a valid height , it will be ambiguous . It could either of the two things
- The distribution never happened 
- The distribution happened at height 1